projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.ts

```diff
@@ -1,122 +1,72 @@
 import { ChangeDetectionStrategy, Component } from '@angular/core';
-import { Product } from '@spartacus/core';
 import { CmsComponentData } from '@spartacus/storefront';
-import { EMPTY, Observable, of } from 'rxjs';
+import { Observable } from 'rxjs';
 import {
   distinctUntilKeyChanged,
   filter,
   map,
   switchMap,
-  withLatestFrom,
 } from 'rxjs/operators';
 import { CmsMerchandisingCarouselComponent } from '../../../cds-models/cms.model';
-import { CdsMerchandisingProductService } from '../../facade/cds-merchandising-product.service';
-import {
-  MerchandisingProduct,
-  MerchandisingProducts,
-} from '../../model/merchandising-products.model';
+import { MerchandisingCarouselComponentService } from './merchandising-carousel.component.service';
+import { MerchandisingCarouselModel } from './model/index';
 
 @Component({
   selector: 'cx-merchandising-carousel',
   templateUrl: './merchandising-carousel.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class MerchandisingCarouselComponent {
+  constructor(
+    protected componentData: CmsComponentData<
+      CmsMerchandisingCarouselComponent
+    >,
+    protected merchandisingCarouselComponentService: MerchandisingCarouselComponentService
+  ) {}
+
   private componentData$: Observable<
     CmsMerchandisingCarouselComponent
   > = this.componentData.data$.pipe(filter(Boolean));
 
+  carouselStyle$: Observable<{
```

Comment:
Instead of making dynamics maybe we can do something simpler like this 

<img width="656" alt="Screen Shot 2019-12-04 at 3 25 06 PM" src="https://user-images.githubusercontent.com/29099466/70178495-643a1180-16aa-11ea-9bec-71eb4079c3ea.png">

Reply:
and in the HTML, something like this

<img width="634" alt="Screen Shot 2019-12-04 at 3 25 26 PM" src="https://user-images.githubusercontent.com/29099466/70178519-6ef4a680-16aa-11ea-9f64-93694927a5b4.png">

Reply:
hmmm, I thought I tried that but had issues because then you need 2 `ngStyle` attributes on the same element. I will try it again though

Reply:
Yes, so this does not work when it is applied to the `cx-carousel`. Do we want to have one approach for the `cx-carousel` (the dynamics) and a normal binding for the individual items or the same approach for both?

Reply:
hmmmmm, I tried that snippet above and got this
<img width="1297" alt="Screen Shot 2019-12-04 at 5 10 25 PM" src="https://user-images.githubusercontent.com/29099466/70186544-1cbb8180-16ba-11ea-90f7-d75e0d0a9211.png">

at carousel level, the background color and title changed
at the product level, the price and name changed to the appropriate ones.

Also, I saw you encapsulated the name and price within a div, but I think it's best to put it in the <a></a> tag

Reply:
Also, can you try to run this in ssr to make sure it works as intended https://sap.github.io/cloud-commerce-spartacus-storefront-docs/server-side-rendering-in-spartacus/#installation-steps-spa-development

Reply:
In ssr mode, there's no browser. Therefore, `document` and `window` object are not available. I presume that the `[style. ...]` bindings that Brian suggested use document under the hood, so we'd like to make sure that it works in ssr mode and that angular doesn't throw an error.

Reply:
Ok, something weird must have been going on with my network connection/GitHub last night as I could only see the first comment about the comopnent ts, not the use of `[style.background-color]`.

Reply:
I tried running in SSR mode with and without the style attributes but the carousel did not load (I did not see any errors in the logs in my terminal window). Does something have to be done to enable a component for SSR mode?